core/primitives/src/block.rs

```diff
@@ -596,6 +607,11 @@ impl Block {
             .0
     }
 
+    pub fn compute_challenges_root(challenges: &Challenges) -> CryptoHash {
+        merklize(&challenges.iter().map(|challenges| challenges.hash).collect::<Vec<CryptoHash>>())
```

Comment:
`... map(|challenge| challenge.hash) ...`

It's a single challenge in the closure, not multiple.

Reply:
++